File: Code/BioMed-DI/DataFusion2/DI3/DI3Datasets.cs
using System;
using System.IO;
using System.Collections.Generic;

namespace DataFusion2
{
    class DI3Datasets
    {


        public static List<FileInfo> datasets()
        {

            List<FileInfo> di3datasets = new List<FileInfo>();
        
            di3datasets.Add(KaessmannDiseaseAssociations);

            for (int i = 1; i <= 50; i++)
            {

                di3datasets.Add(gene2pubtatorcentral(i));

            }

            return di3datasets;

        }

        public static FileInfo KaessmannDiseaseAssociations = new FileInfo(Environment.CurrentDirectory + "/data/input/DI3/DI2_dt.xml");

        public static FileInfo gene2pubtatorcentral(int i) 
        {
            return new FileInfo(Environment.CurrentDirectory + "/data/input/DI3/gene2pubtatorcentral_" + i + "_dt.xml");
        }

    }

}

File: Code/BioMed-DI/DataFusion/src/main/java/genes/DataFusion/solution/DI3/DI3Datasets.java
package genes.DataFusion.solution.DI3;

// java
import java.io.File;

// winter
import de.uni_mannheim.informatik.dws.winter.model.FusibleDataSet;
import de.uni_mannheim.informatik.dws.winter.model.FusibleHashedDataSet;
import de.uni_mannheim.informatik.dws.winter.model.defaultmodel.Attribute;

// model
import genes.DataFusion.model.Gene.Gene;
import genes.DataFusion.model.Gene.GeneXMLReader;

public class DI3Datasets {


    public static FusibleDataSet<Gene, Attribute> gene2pubtatorcentral(int i) throws Exception {

        FusibleDataSet<Gene, Attribute> ds = new FusibleHashedDataSet<>();

        new GeneXMLReader().loadFromXML(new File("data/input/DI3/gene2pubtatorcentral_" + i + "_dt.xml"), "/genes/gene", ds);

        return ds;

    }
    
}
